Automattic/jetpack
# Add/contextualized jetpack connection

#### Changes proposed in this Pull Request:

* Adds a new screen that's shown when the Jetpack Dashboard of a site is shown for the first time. This screen appears for both connected and disconnected sites, and also appears when somebody activates Jetpack directly from Woo, or the plugins page.
See the screenshots below:

| Connected | Disconnected |
|-----------|--------------|
|![image](https://user-images.githubusercontent.com/5714212/143513940-9785a965-884f-4513-b1ac-f9a41099da19.png)|![image](https://user-images.githubusercontent.com/5714212/143514024-6bfd13f5-bc4f-4d54-b142-e59eedd26150.png)|

#### Jetpack product discussion


Context: pbNhbs-1c9-p2

#### Does this pull request change what data or activity we track or use?


- A new track was added to record when a user clicks the "Continue to Jetpack" button. The event name is `jetpack_wpa_click` and its `target` is `contextualized_connection_continue_button`

#### Testing instructions:






* Spin up a site with this Jetpack branch, or use the live link, or a local installation
* Install and activate WooCommerce
* With WooCommerce active, click on Jetpack on the left menu to go to the Jetpack Dashboard
* Make sure you are redirected to `/wp-admin/admin.php?page=jetpack#/woo-setup` and the screen you see matches the screenshots above based on the Jetpack connection status

There are a few other scenarios where you can get to that screen (WooCommerce always need to be installed and active):
* If you already have Jetpack, but have never seen that screen before
* If you have just installed Jetpack, connected through any screen, but visited the Dashboard for the first time
* If you don't have Jetpack installed, and then installs it through WooCommerce setup (after installing WooCommerce, you'll be prompted to install Jetpack either from the Woo installation flow, or afterwards on the Woo Dashboard or Woo Marketplace)
* If you don't have Jetpack and install it from the WordPress Plugin page
